### ISSUE TITLE
[Bug] Client Credential should not be allowed over tenantless authorities

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#1384]
**Which Version of MSAL are you using ?**
4.3.x

**What authentication flow has the issue?**
Client creds

**Repro**

1. Create CCA with tenantless AAD authority (e.g. common) 
2. AcquireTokenForClient  -> fetches a token, for tenant T1
3. AcquireTokenForClient  with authority override set tenanted authority T1

**Actual:** Second call to AcquireTokenForClient should fetch the AT in the cache
**Expected:** MSAL talks to Evo to get a new token

Note: discussed this with Adrian Frei and Evo's stance is that client_credentials should never have he authority set to `common`. From MSAL perspective, we cannot get the actual tenant ID because we normally get it from the ID Token, which is not present in response.

**Comment by jmprieur** [comment]
@hpsin : can you please confirm that we do want to disallow `organizations` in Client credentials in next major version of MSAL.NET

**Comment by bgavrilMS** [comment]
(and `common` as well)

**Comment by jmprieur** [comment]
common for sure (it does not make sense for client creds)

**Comment by hpsin** [comment]
Yes - `common` and `organizations` are both aliases over multiple tenants, and do not specify which tenant a token should be requested for.  For security and reliability, apps should explicitly indicate what tenant they want to get an app-only token for instead of having AAD do best-effort guesses.

**Comment by hpsin** [comment]
`consumers` is ok, as app tokens are valid there and `consumers` is a single tenant. Please do not block use of `consumers`.

**Comment by jmprieur** [comment]
Thanks @hpsin

**Comment by bgavrilMS** [comment]
I'd like to take this as a non-breaking change. It looks like ESTS-R blocks it anyway.

**Comment by maliksahil** [comment]
We should go forward with this, but can we start with a warning in a major version, before an error or disallow in the next?

**Comment by bgavrilMS** [comment]
We already log an error message for this case, and this has been the case for a while.
